lib/Map/Vector/Protomaps/ProtomapsGeojsonSource.ts

```diff
@@ -0,0 +1,318 @@
+      ) {
+        featureInfo.position = Cartographic.fromDegrees(
+          f.geometry.coordinates[0],
+          f.geometry.coordinates[1]
+        );
+      }
+
+      featureInfo.configureDescriptionFromProperties(f.properties);
+      featureInfo.configureNameFromProperties(f.properties);
+
+      featureInfos.push(featureInfo);
+    });
+
+    return featureInfos;
+  }
+}
+
+export const geomTypeMap = (
+  type: string | null | undefined
+): GeomType | null => {
+  switch (type) {
+    case "Point":
+    case "MultiPoint":
+      return GeomType.Point;
+    case "LineString":
+    case "MultiLineString":
+      return GeomType.Line;
+    case "Polygon":
+    case "MultiPolygon":
+      return GeomType.Polygon;
+    default:
+      return null;
+  }
+};
+
+export function geojsonVtTileToProtomapsFeatures(
+  features: geojsonvt.Feature[],
+  tileSize: number
+): ProtomapsFeature[] {
+  const scale = tileSize / GEOJSON_VT_EXTENT;
+
+  return features
+    .map((f) => {
+      let transformedGeom: Point[][] = [];
+      let numVertices = 0;
+
+      // Calculate bbox
+      const bbox: Bbox = {
+        minX: Infinity,
+        minY: Infinity,
+        maxX: -Infinity,
+        maxY: -Infinity
+      };
+
+      // Multi-polygon
+      if (Array.isArray(f.geometry[0][0])) {
+        // Note: the type is incorrect here
+        const geom = f.geometry as unknown as [number, number][][];
+        transformedGeom = geom.map((g1) =>
+          g1.map((g2) => {
+            const x = g2[0] * scale;
+            const y = g2[1] * scale;
+
+            if (bbox.minX > x) {
+              bbox.minX = x;
+            }
+
+            if (bbox.maxX < x) {
+              bbox.maxX = x;
+            }
+
+            if (bbox.minY > y) {
+              bbox.minY = y;
+            }
+
+            if (bbox.maxY < y) {
+              bbox.maxY = y;
+            }
+            return new Point(x, y);
+          })
+        );
+        numVertices = transformedGeom.reduce<number>(
+          (count, current) => count + current.length,
+          0
+        );
+      }
+      // Other feature types
+      else {
+        const geom = f.geometry as [number, number][];
+        transformedGeom = [
+          geom.map((g1) => {
+            const x = g1[0] * scale;
+            const y = g1[1] * scale;
+
+            if (bbox.minX > x) {
+              bbox.minX = x;
+            }
+
+            if (bbox.maxX < x) {
+              bbox.maxX = x;
+            }
+
+            if (bbox.minY > y) {
+              bbox.minY = y;
+            }
+
+            if (bbox.maxY < y) {
+              bbox.maxY = y;
+            }
+            return new Point(x, y);
+          })
+        ];
+        numVertices = transformedGeom.length;
+      }
+
+      if (f.type === 0) return null;
+
+      const geomType = {
+        [1]: GeomType.Point,
+        [2]: GeomType.Line,
+        [3]: GeomType.Polygon
+      }[f.type];
```

Comment:
I think commit e32e129e7bb addresses the same issue without resorting to comparing to the raw values in the enum.